goperf: init at unstable-2023-11-08

pbsds opened the pull request:
###### Description of changes

Project page: <https://cs.opensource.google/go/x/perf>

The package name is inspired by the package `gotools`.

This module contains command-line tools for analyzing go benchmark result data. Of interest are:

* `benchstat`, which computes statistical summaries and A/B comparisons of Go benchmarks.
* `benchsave`, which publishes benchmark results to <https://perf.golang.org>

The contents of `result`:
```
-r-xr-xr-x 2 root root 9,7M jan.   1  1970 result/bin/appengine
-r-xr-xr-x 2 root root 6,6M jan.   1  1970 result/bin/benchsave
-r-xr-xr-x 2 root root 9,0M jan.   1  1970 result/bin/benchseries
-r-xr-xr-x 2 root root 3,0M jan.   1  1970 result/bin/benchstat
-r-xr-xr-x 2 root root 9,1M jan.   1  1970 result/bin/localperf
-r-xr-xr-x 2 root root 7,9M jan.   1  1970 result/bin/localperfdata
```

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

pbsds committed:
goperf: init at unstable-2022-09-20

pbsds commented:
@ofborg build

jfvillablanca commented:
Result of `nixpkgs-review pr 196196` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>goperf</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1179

pbsds committed:
goperf: unstable-2022-09-20 -> unstable-2023-11-08, add updateScript

the unstable git updater does not bump the go-module vendor hash, hence the crimes
